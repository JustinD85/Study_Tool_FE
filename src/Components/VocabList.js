import React, { Component } from 'react';
import uuid from 'uuid';

class VocabList extends Component {

  changeToJSX(word) {
    const { handleSelectWord, currentWord } = this.props;
    return (<li
      onClick={(e) => handleSelectWord(e.target.innerText)}
      key={uuid()}
      className={currentWord === word ? 'selected-word' : undefined}
    >
      {word}
    </li>)
  }

  returnVocabAsLi = () => {
    const { currentTopic, topics, user } = this.props;
    let currentVocab = [];

    topics.get(currentTopic).forEach((section) => {
      section.get('vocabulary').forEach((definition, word) => {
        if (!user[currentTopic].words.includes(word)) currentVocab.push(word);
      });
    });

    return currentVocab.sort().map(word => this.changeToJSX(word));
  }

  render() {

    return (
      <div className="vocab-container">
        <h2>Word List</h2>
        <ul className="vocab-list">
          {this.returnVocabAsLi()}
        </ul>
      </div>
    )
  }
}

export default VocabList;