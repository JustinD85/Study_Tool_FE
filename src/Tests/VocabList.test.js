import React from 'react';
import VocabList from '../Components/VocabList';
import { shallow } from 'enzyme';

describe('<VocabList/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <VocabList />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});