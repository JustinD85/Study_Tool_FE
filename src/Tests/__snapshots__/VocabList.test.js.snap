// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VocabList/> should match the snapshot 1`] = `undefined`;
